packages/lib/services/searchengine/SearchEngine.ts

```diff
@@ -554,6 +554,94 @@ export default class SearchEngine {
 		return SearchEngine.SEARCH_TYPE_FTS;
 	}
 
+	makeSearchTree(str: string) {
+		// (tag:A (any:1 tag:B tag:C)) -> [ "tag:A", [ "any:1", "tag:B", "tag:C"]]
+		let i = 0;
+		function main(): any {
```

Comment:
Instead using "any", please create a new interface that describe the type. In fact, to improve type checking it might be better to create an interface like `interface Node { content: string, children: Node[] }`. That way you don't need to use an array with mixed content.